addons/website/static/src/js/content/menu.js

```diff
@@ -905,7 +905,9 @@ publicWidget.registry.HeaderGeneral = publicWidget.Widget.extend({
      */
     _onCollapseHidden() {
         this.options.wysiwyg?.odooEditor.observerUnactive("removeCollapseClass");
-        this.el.classList.remove('o_top_menu_collapse_shown');
+        if (!this.el.querySelector("#top_menu_collapse_mobile.show")) {
```

Comment:
```suggestion
        const mobileNavbar = this.el.querySelector("#top_menu_collapse_mobile");
        if (!mobileNavbar.classList.contains("show") && !mobileNavbar.classList.contains("showing")) {
```

Since https://github.com/odoo/odoo/commit/058212e12b5079eba870bde9775fe98f27928935 and the update to Bootstrap 5.3.3, there's a new `.showing` class on offcanvas menu.